Sanitize untrusted prompt input (agent-input-sanitizer) (#367)
brendanlong/clawed-burrow: doc/DESIGN.md

- `bypassPermissions` mode is used since the machine is dedicated to running this app
- The machine should be dedicated to this application — not shared with other users

### Input Sanitization

> Untrusted text is scrubbed before it reaches the model, defending against hidden-content prompt injection (invisible Unicode, ANSI escapes, human-invisible HTML comments/elements) and surfacing data-exfil-shaped URLs. This uses the [`agent-input-sanitizer`](https://github.com/alexander-turner/agent-input-sanitizer) package (the standalone, importable piece of [claude-guard](https://github.com/alexander-turner/claude-guard) — the sandbox/VM and egress-firewall layers are intentionally **not** adopted because this host deliberately grants broad capability, e.g. RunPod/GPU access).

Author: We haven't made a decision on claude-guard so let's leave that out.

```suggestion
Untrusted text is scrubbed before it reaches the model, using  [`agent-input-sanitizer`](https://github.com/alexander-turner/agent-input-sanitizer), defending against hidden-content prompt injection (invisible Unicode, ANSI escapes, human-invisible HTML comments/elements) and surfacing data-exfil-shaped URLs.
```

Author: Applied your suggested wording in cb36dfe, and also scrubbed the other claude-guard framing in that section (the "intentionally not adopted" parenthetical). The adoption question is now tracked separately in #377.

— Claude
